File: math/factorial.go
package main

import "fmt"

// Time complecity Θ(n)
// Auxiliar space complexity Θ(1)
func factorial_loop(n int) int {
	factorial := 1
	for i := n; i >= 1; i-- {
		factorial *= i
	}
	return factorial
}

// Time complecity Θ(n)
// Auxiliar space complexity Θ(n)
func factorial_recursive(n int) int {
	if n <= 1 {
		return 1
	}
	return n * factorial_recursive(n-1)
}

func main() {
	fmt.Println("Factorial of 5 is ", factorial_loop(5))
	fmt.Println("Factorial of 6 is ", factorial_recursive(6))
}

File: math/factorial.c
#include <stdio.h>

// Time complecity Θ(n)
// Auxiliar space complexity Θ(n)
int factorial_recursive(int n) {
  if (n <= 1)
    return 1;
  return n * factorial_recursive(n - 1);
}

// Time complecity Θ(n)
// Auxiliar space complexity Θ(1)
int factorial_loop(int n) {
  int result = 1;
  for (int i = n; i > 1; i--) {
    result *= i;
  }

  return result;
}

int main() {
  printf("Factorial of 1 is %d\n", factorial_loop(1));
  return 0;
}